src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -370,87 +371,89 @@ private fun processQualifiedPathResolveVariants(
     parent: PsiElement?,
     processor: RsResolveProcessor
 ): Boolean {
-    val (base, subst) = qualifier.reference?.advancedResolve() ?: run {
+    val variants = qualifier.reference?.advancedMultiResolve()?.notEmptyElseNull() ?: run {
         val primitiveType = TyPrimitive.fromPath(qualifier)
         if (primitiveType != null) {
             if (processTypeQualifiedPathResolveVariants(lookup, path, processor, ns, primitiveType, null)) return true
         }
         return false
     }
-    val isSuperChain = isSuperChain(qualifier)
-    if (base is RsMod) {
-        val s = base.`super`
-        // `super` is allowed only after `self` and `super`
-        // so we add `super` in completion only when it produces valid path
-        if (s != null && (!isCompletion || isSuperChain) && processor("super", s)) return true
+    for ((base, subst) in variants) {
```

Comment:
Hmm, why this multiresolve logic is needed to resolve enum items by Self?

Reply:
If I understand your question right, it is because `Self` was previously resolved to `impl` item, now it is resolved to both `impl` and `enum` items.

Reply:
Ok, I get it. We probably can make it simpler. Instead of resolution `Self` into 2 places, we can resolve it to `impl` only ane then handle that `impl` in `ItemResolution.kt` `processItemOrEnumVariantDeclarations` near `RsTypeAlias` case

Reply:
Am I understand it right, we should turn enum varinats into associated items, like static functions?

Reply:
Umm, yes, something like that. Maybe I'm wrong with `ItemResolution.kt` and really it should be handled in `processQualifiedPathResolveVariants` next to the other `impl`/`Self`-related stuff.

Btw it would be nice to test a corner case when there is an enum variant and an associated function with the same name.

Reply:
Tried to follow your suggessions, seems working

Reply:
>Btw it would be nice to test a corner case when there is an enum variant and an associated function with the same name.
Currently in both cases, with enum name and with `Self`, resolves to both enum variant and method with the same name, so I think it should be separate issue and must include some inspection to give overlap error. Interesting that rustc doesn't report it as error yet https://github.com/rust-lang/rust/issues/48758